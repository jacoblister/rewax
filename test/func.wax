; waxc - nothing output 
; (func func_no_body
; )

(func func_empty 
  (print "func_empty:")
)

(func func_result (result int)
  (print "func_result:")
  (return -1)
)

(func func_param_a (param a int)
  (print "func_param_a:")
  (print a)
)

(func func_param_abc (param a int) (param b int) (param c int)
  (print "func_param_abc:")
  (print a)
  (print b)
  (print c)
)

(func func_param_abc_result (param a int) (param b int) (param c int) (result int)
  (print "func_param_abc_result:")
  (print a)
  (print b)
  (print c)
  (return -1)
)

(func func_str_result (result str)
  (print "func_str_result:")
  (return "string")
)

(struct point 
    (let pos_x float)
    (let pos_y float)
)

(func func_struct (param p (struct point))
  (print "func_struct:")
  (print (get p pos_x))
  (print (get p pos_y))
)

(func main (result int)
    ; (call func_no_body)
    (call func_empty)
    (print (call func_result))
    (call func_param_a 1)
    (call func_param_abc 1 2 3)
    (print (call func_param_abc_result 1 2 3))
    (print (call func_str_result))

    (let p (struct point) (alloc (struct point)))
    (set p pos_x 1.1)
    (set p pos_y 2.2)
    (call func_struct p)
    (return 0)
 )
