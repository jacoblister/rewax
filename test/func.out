func_empty:
func_result:
-1
func_param_a:
1
func_param_abc:
1
2
3
func_param_abc_result:
1
2
3
-1
func_str_result:
string
func_struct:
1.1
2.2
